drivers/lora/sx1276.c

```diff
@@ -156,6 +157,13 @@ void DelayMsMcu(uint32_t ms)
 	k_sleep(ms);
 }
 
+static void sx1276_dio_work_handle(struct k_work *work)
+{
+	int dio = work - dev_data.dio_work;
```

Comment:
This one looks odd but I don't have any better solution :)

Reply:
One strategy is to define a struct containing a `k_work` item and an `ID` (or even directly the pointer to the `DioIRQ` function to be called):

```
struct dio_work_t {
        struct k_work work;
        int dio;
};
```
This `struct dio_work_t` can be used in `sx1276_data` instead of the `struct k_work`. Then `sx1276_dio_work_handle` can be written that way:

```
static void sx1276_dio_work_handle(struct k_work *work)
{
        struct dio_work_t *w = CONTAINER_OF(work, struct dio_work_t, dio_work);
        (*DioIrq[w.dio])(NULL);
}

All that said I don't think it really worth it the change, I think the current code is fine.
```